### ISSUE TITLE
Improvements to Assets page (i.e. show DEX total volume, etc.)

### ISSUE THREAD

**Issue by tbone-bts** [oxarbitrage/bitshares-explorer#2]
On the assets page, these would be nice improvements:

1. Show total DEX volume at the top
2. Show all assets volume in USD terms (perhaps down the road give choice of currency to user)
3. Daily chart showing DEX volume (maybe this needs to be on another page)
4. Fix asset volume numbers (seem to be off by an order of magnitude?)
5. Enable sorting by 24-hour volume and also by Market cap (clickable column heads?)

Thanks!

**Comment by tbone-bts** [comment]
P.S.  cryptofresh seems to be down permanently now, so perhaps you can consider highly prioritizing display of the 24-hour volume column in terms of a common currency.  Since that column is currently based on each asset's volume (regardless of value), we are getting very irrelevant assets at the top, while the more serious assets are much less visible.  

It would be nice to have the choice of display currency for this column (e.g. BTS, BTC, or choice of fiat), but in the interest of making a quick/important improvement, perhaps consider displaying it in terms of BTS for now?  It would help a lot.

**Comment by oxarbitrage** [comment]
got it, i am trying to finish some work in the bitshares-core today. ill get back to the explorer tomorrow and work on it the rest of the week. will go over this issue first.

**Comment by tbone-bts** [comment]
It would be great if the Assets page could look something like this page from CoinMarketcap.  The main differences between this and the Assets page on the explorer are a) the sorting by base asset, b) ability to choose base asset (BTS, BTC, USD, CNY, etc), c) showing total volume at the top, and d) the asset icons (not critical, but a nice touch).  

Obviously the most important thing is sorting by base asset (and showing total volume at the top).  That will make this page a useful tool that people will refer to multiple times per day.  And it will get assets like compuceeds off the top row and instead show the assets with real volume at the top.  I really think this is critical.  Hope we can get it soon.  Thanks


![bts_explorer_assets_page](https://user-images.githubusercontent.com/15830348/30805441-65b54852-a1bf-11e7-958f-d725c5a02a8c.JPG)

**Comment by tbone-bts** [comment]
By the way, I noticed that BTS token is missing from the Assets page.

**Comment by oxarbitrage** [comment]
related to bts missing in assets list: https://github.com/oxarbitrage/bitshares-explorer/issues/7

working on the assets page, top priority in the explorer. ill update you as soon as i have something.

**Comment by tbone-bts** [comment]
Really nice progress on this today, Alfredo!  Here is some additional input for your consideration:

1. Are you able to distinguish between **gateway** UIAs (i.e. UIAs for assets not native to Bitshares) vs. UIAs that are assets **native** Bitshares?  If so, can you indicate that in the "Type" column?  Like maybe "User Issued (Native)" and "User Issued (Gateway)".

2. Values in the "Market Cap" and "Supply" columns are not relevant for **non-native** Bitshares assets and actually cause distraction in this view.  Can you display these values only for **native** Bitshares assets and leave them blank (or use "-") for the **non-native** assets?  (Note: I would continue to display these values on the asset **detail page**, perhaps with a notation indicating the value doesn't represent that coin's supply on its own blockchain).

3. Can you modify the column order?  Icon, Name, Price (BTS), 24h Volume, Type, Market Cap, Supply, Holders

4. Can you right justify numeric values (and standardize # of decimal places so everything lines up)?

5. Some Smartcoins are missing including bitUSD, bitRuble, bitCAD, bitGBP, bitJPY, bitSilver, HERO

6.  Can you reduce the row height so more assets fit within view on the page?

This is really shaping up.  Thanks for your effort!

**Comment by oxarbitrage** [comment]
Thanks @tbone-bts , let me ask you in regards to 1:

Pls send sample of gateway and native uia. i am not sure if i can distinguish them without actually flag them manually as being native/gateway. For example is blockpay a gateway uia ? ppy ? how do we know ?

In regards to point 5, all the smartcoins you mentioned are in the list except for the ones with no volume in the last 24 hs. HERO is in the list, SILVER, USD, RUBLE, etc. The "bit" prefix seems to be something some clients add to smartcoins, the core does not name them that way and it is just USD instead of bitUSD. GBP, JPY and others are probably missing because of no volume in the last 24 hours.

**Comment by oxarbitrage** [comment]
for reference, the asset pages changes we are working on is at http://bitshares-explorer.io:9009/#/assets

**Comment by tbone-bts** [comment]
>Pls send sample of gateway and native uia. i am not sure if i can distinguish them without actually flag them manually as being native/gateway. For example is blockpay a gateway uia ? ppy ? how do we know ?

Blockpay is a native UIA.  PPY is a gateway UIA (i.e. it's native to another blockchain and requires gateway support).  More examples of gatweway UIAs are: BTC, LTC, ETH, DASH, MAID, LISK, PPC, etc.  Additional examples of native UIAs are: OBITS, ICOO, BTSR, KEXCOIN, BEYONDBIT, BADCOIN, etc.  I guess there can't really be an asset property on the blockchain that makes this distinction, eh?  But the distinction is made for the UI Deposit/Withdraw, so there must be some way.  Or are they just doing it manually? 

>In regards to point 5, all the smartcoins you mentioned are in the list except for the ones with no volume in the last 24 hs. HERO is in the list, SILVER, USD, RUBLE, etc. The "bit" prefix seems to be something some clients add to smartcoins, the core does not name them that way and it is just USD instead of bitUSD. GBP, JPY and others are probably missing because of no volume in the last 24 hours.

Hmmm, I'm pretty sure all of the ones I listed were not visible before.  Now some of them **are** visible.  But not all.  GBP, CAD and JPY are still missing.  

>for reference, the asset pages changes we are working on is at http://bitshares-explorer.io:9009/#/assets

Yes, this is where I'm viewing changes.  :)

**Comment by tbone-bts** [comment]
Two things I forgot:

1.  Can you adjust the default sort to be **descending by Volume**? (this is critical, I can't believe I forgot it!)
2.  For assets with no icon, can you leave it blank instead of using a default image?

**Comment by oxarbitrage** [comment]
GBP: http://23.94.69.140:5000/get_volume?base=BTS&quote=GBP -> 0 volume
JPY: http://23.94.69.140:5000/get_volume?base=BTS&quote=JPY -> 0 volume
CAD: http://23.94.69.140:5000/get_volume?base=BTS&quote=CAD -> 0 volume

compare with: http://23.94.69.140:5000/get_volume?base=BTS&quote=CNY

assets with 0 volume against bts do not display because if i do so i will have a table with thousands of assets. as they start to get volume they will show up on page.

**Comment by tbone-bts** [comment]
Ah, ok.  Now I see that you already mentioned it in your previous post.  Somehow I missed that.  But yeah, that makes perfect sense.

**Comment by oxarbitrage** [comment]
what do you think on removing the market cap and supply columns fully in the assets page as they cxan still see this data inside the assets page ?

**Comment by tbone-bts** [comment]
Actually, I really think those columns are very valuable to see for the native assets.

**Comment by oxarbitrage** [comment]
ok, lets leave it then. i made a few changes more:
- removed default icon to white.
- reordered columns.
- reduced icon size from 50 to 30px.
- right aligned numbers columns.

**Comment by oxarbitrage** [comment]
the order was changed so the asset with more volume is at the top. this among the other changes we did was enough to move the new assets page to the production version here: http://bitshares-explorer.io/#/assets

in regards to the volume figures of the top, please note this is total volume of bts markets. this should be close to another proposed total volume metrics proposed that was top 30. the real total DEX volume will be the numbers we are showing plus all the markets volume where BTS in not part of the market for example CNY/YOYOW. this volume is not computed right now due to the difficulties it haves to express this volume in bts terms. when i figure out this, i will get back to this page and change volumes figures of the top to real total volume.

the volume chart is made with the daily "24HS VOLUME IN BTS". current data is test data but the chart will start to have more sense as the days pass and real data is plotted.

**Comment by tbone-bts** [comment]
@oxarbitrage It looks for the Deposit/Withdraw page in the UI, they identify which assets are "gateway" assets using the json in the link below (where walletType = bitshares2).  This is only for OL assets, but I assume RUDEX will provide something similar once they are ready to have their gateway listed in the UI.  Can you use this to a) remove the values from the "Market Cap" and "Supply" columns of these assets and b) be more specific in the Type column i.e. "User Issued (Native)" and "User Issued (Gateway)"?

https://ol-api1.openledger.info/api/v0/ol/support/coins

**Comment by tbone-bts** [comment]
I would like to offer some additional feedback to help make this table as readable and therefore as useful as possible.  

Below are 2 mock-ups that incorporate some of the previously discussed things: **a)** denoting Gateway vs. Native UIA Types, **b)** removing "Market Cap" and "Supply" values for User Issued (Gateway) Types.  

Also, some additional things included in the mock-ups to make the table more user-friendly: 

**1)** it's far more readable when the currency units are displayed with the values in the table (as CMC and Cryptofresh do) instead of in the column headers.  Fiat currency symbol (i.e. $, ¥, €, £, ₩, etc.) can go before the value, and crypto-currency symbols (i.e. BTS, BTC) can go after the value (as CMC does).  

**2a)** currently it's difficult to see which column is being sorted.  It's also difficult to select the little tiny up or down arrow in order to choose sort column and the sort order (asc or desc).  Can you make the column names clickable for sorting where clicking it once sorts descending, clicking again sorts ascending?  
**2b)** A box around the sorted column (or a slightly shaded background for that column) would make it easy to see which column is being sorted.  
**2c)** It's also a nice touch to add a little arrow next to the column header to further denote which column is sorted, and also show the direction.  

with fiat currency symbol:
![bts_explorer_assets_page_layout_2_](https://user-images.githubusercontent.com/15830348/31103365-8d60add4-a7a4-11e7-8348-085efd455898.jpg)

with crypto-currency symbol:
![bts_explorer_assets_page_layout_2_bts](https://user-images.githubusercontent.com/15830348/31103359-8a62344a-a7a4-11e7-9c1e-6b6f5b9d3d93.jpg)

**Comment by oxarbitrage** [comment]
> Below are 2 mock-ups that incorporate some of the previously discussed things: a) denoting Gateway vs. Native UIA Types, b) removing "Market Cap" and "Supply" values for User Issued (Gateway) Types.

As discussed in telegram here we do the filter if the asset is open.xxx or rudex.xxx. check http://bitshares-explorer.io/#/assets we can flag additional coins manually if needed.

> 1) it's far more readable when the currency units are displayed with the values in the table (as CMC and Cryptofresh do) instead of in the column headers. Fiat currency symbol (i.e. $, ¥, €, £, ₩, etc.) can go before the value, and crypto-currency symbols (i.e. BTS, BTC) can go after the value (as CMC does).

done.

> 2a) currently it's difficult to see which column is being sorted. It's also difficult to select the little tiny up or down arrow in order to choose sort column and the sort order (asc or desc). Can you make the column names clickable for sorting where clicking it once sorts descending, clicking again sorts ascending?

done.

> 2b) A box around the sorted column (or a slightly shaded background for that column) would make it easy to see which column is being sorted.

not easy, i had been trying since yesterday. i understand the benefit of being down in the table knowing what order you are currently on but not much more to invest more time on it. will not be done by now.

> 2c) It's also a nice touch to add a little arrow next to the column header to further denote which column is sorted, and also show the direction.

done.

**Comment by tbone-bts** [comment]
Moved this from https://github.com/oxarbitrage/bitshares-explorer/issues/11#event-1293386998 per request by @oxarbitrage: 

The assets page has had tremendous progress. Thanks for that! There are still a few items and I've been meaning to leave more feedback, just been so busy. Can it be reopened, or does a new one need to be started? Here are some notes:

1. I think the chart value is off by an order of magnitude.. 

2. the distinction between gateway UIAs vs. native UIAs you added is very helpful. can you also reflect that in the Type column? i.e. User Issued (Native) vs. User Issued (Gateway)?

3. this page has become very useful. that usefulness could really be expanded by giving user ability to switch from BTS as base currency to fiat of their choice. to me personally USD is most useful
 
4. can you make any reference to 24 hour "24h" instead of "24 hs"? in the U.S. we commonly use "24hr", and looks like you guys use "24 hs"....but I think "24h" is more universal and cleaner.

5. The values in the upper left need some work. i made some notes several days ago that i didn't get to finish and now i can't find them. but to start, the market cap values shouldn't say 24h, only the volumes. also, BTS market cap doesn't make sense in BTS terms. market cap should be in terms of BTC, USD, and CNY. although one other thing i'm noticing, i'm not even sure those "market cap" numbers are really the market cap (i.e. total value of outstanding BTS tokens). is that what you intended it to be?

6. i see the line around the column that is sorted is too much trouble. changing the background might be easier. although honestly, that is not really necessary. other items are much more important. plus now that you have an arrow only on the sorted column, it's obvious enough which column is sorted. i would just remove the line since it's not really working as intended.

**Comment by tbone-bts** [comment]
from @oxarbitrage, moved from: https://github.com/oxarbitrage/bitshares-explorer/issues/11#event-1293386998

>please move this comments to the assetS issue (#2) or open new one. This is the issue for the individual asset page. closing again.

>just for your consideration from the points posted here some of them will not be done at least by now.

>1- pls explain more, i am not understanding the "off by 1 magnitude thing"
3- will not be done now, too hard.
6- whatever you mean there will not be done.

>working on the rest of them now.

Thanks Alfredo! 

Re: 1, it seemed the total DEX volume was 10x higher than the volume from Cryptofresh.  At least that's what it looked like on Oct. 11 when the total DEX volume on Cryptofresh was about $6M and your chart was showing $60M.  But now the difference is smaller.  On Cryptofresh today the volume is $4.2M, and on your explorer it's $8.8M.  

Re: 3, I understand it may be a lot more work to add a drop-down to enable user choice.  In that case, what do you think about just hard-coding USD for now since it's more useful than seeing the values in terms of BTS, and it would enable us to more easily compare the values to Cryptofresh?   

Re: 6,  I was basically just saying (a) it seems like too much work to put the line around the entire sorted column (like in the mock-up), and (b) it seems unnecessary to have a line around just the header since now there is an arrow next to **only** the sorted field (as opposed to having arrows next to all headers previously) so it's easy enough now to see which column is sorted. 
 Also, that line around the header is causing some odd behavior, making the columns jump around when you click the headers, so maybe it's best to remove it.

**Comment by oxarbitrage** [comment]
Hi @tbone-bts thanks for taking the time of moving the suggestions into this issue. I really like your suggestions and the assets page looks already a lot better than it was thanks to them. really appreciated. 

The problem is sometimes it is hard to follow the community and their suggestions as they came with all kind of things, all kind of charting they want, etc. I try to follow them but it is hard without some kind of filter of at least who is asking and what it is the position on the community the asker haves in order to follow or i will just drive crazy going after all of them. This is probably something the UI team suffers more than i do. I want to leave clear that this is not your case, your suggestions will be all considered and all be made if possible. 

With that said, from the last points stated the hardest one should be the ability to change the values to the currency of choice. having everything in usd will be easier but i am afraid that will work for you and me until someone came in asking for other currencies. So i am thinking on taking the time to add USD, CNY and EUR and we will be covered. So i am working on that.

In regards to the volume the older bars more in the left are wrong values, they came from some testing i made in those days when i was trying to include volume from all the crosses instead of just BTS crosses. Please note that the volume we are displaying now is the volume of  BTS crosses but leaves outside important markets like CNY/USD or whatever where BTS is not involved.
My idea was to include those markets in the total volume count and that to me will be the real total volume of the DEX but i had a hard time normalizing so i ended up with the bts crosses only. This may be changed if i can get the right math to do it.

Another thing that will change in the future thanks to some improvements we are making in the core is the ability to get this figures in real time, right now, just as cryprofresh we are getting total volume figures once a day. But lets leave that by now as i am not sure how much time is going to be needed to be there.

Anyways, keep the suggestions coming, thank you very much and ill update you as soon as i have something done.

**Comment by o5j5vg55bv5hv5j5f8799f9** [comment]
I've noticed the biggest differences in dex volume came during period of significant margin calls - cryptofresh would show them all typically. I don't know if those volumes are included in this one.

**Comment by tbone-bts** [comment]
@o5j5vg55bv5hv5j5f8799f9 - not sure, but I think @oxarbitrage was going to be looking into possible discrepancies in volume between his explorer and Cryptofresh.  

@oxarbitrage - by the way, I noticed some strange numbers in the Supply and Market Cap columns.  For example, Stealth UIA shows 100,000,000,000 supply and 9,600,015,360,020 BTS market cap.  But those numbers should be 1,000,000 and ~96,000,000 BTS respectively.  Same with BTS core token.  It shows supply of 260,087,384,718,124 and market cap of 260,087,384,718,124 BTS.  Both numbers should be ~2,600,873,847.  So those values are all too high by a factor of 100,000.  But then there's USD which is showing a supply of $32,018,727,442.  That should actually be ~$3,200,000.  So that's off by a factor 10,000. 

One other thing, in order to make the values in the Price column more readable, I wonder if you can: a) make the numbers after the decimal a slightly smaller font, b) make all values have the same number of digits after decimal point, and c) make the decimal point and the commas bold text.   These changes would make that column much more readable.  

Thanks for your consideration and for your effort!
